### PR TITLE
Adjusting CAIP-171 as per last meeting

### PR DESCRIPTION
**Pull request by bumblefudge**
- generalize "token" to "identifier" to be more platform-neutral
- made assumptions more explicit
- add stability assumption (identifier doesn't change as session state changes)

NB @ritave  (can't tag you for review directly)

partly addresses #141 , although #170 is probably the place to further specify session management for shared assumpions.  ideally the choice to use #170 session **objects** would be separate from the choice to use #171 session **identifiers**, and wallets or users would know what they can assume based on one or both of the standards being conformed to?

### CODE REVIEW

File: CAIPs/caip-171.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+`required` front-matter property, it SHALL be interpreted as reference to this
+specification.
+
+> Notice that there are no code constraints on the token value except
+> that it be expressed as a string with few reserve characters, under 256
+> characters in length. This is by design and the value is
+> implementation-dependent.
+
+```typescript
+type SessionToken = string;
```
Comment (ritave) [inline CAIPs/caip-171.md:50]:
```suggestion
type SessionIdentifier = string;
```

File: CAIPs/caip-171.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+```
+
+Properties of the `sessionIdentifier` are as follows:
```
Comment (ritave) [inline CAIPs/caip-171.md:53]:
```suggestion
Properties of the `SessionIdentifier` are as follows:
```

File: CAIPs/caip-171.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### Definition
+
+Whenever a CAIP uses a name `SessionIdentifier` and has this CAIP in the
+`required` front-matter property, it SHALL be interpreted as reference to this
+specification.
+
+> Notice that there are no code constraints on the token value except
+> that it be expressed as a string with few reserve characters, under 256
```
Comment (ritave) [inline CAIPs/caip-171.md:45]: Remove the size limit from a non-normative section and let's put it into specification below

File: CAIPs/caip-171.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+```typescript
+type SessionToken = string;
+```
+
+Properties of the `sessionIdentifier` are as follows:
+1. It MUST uniquely identify an open and stateful session. 
+1. It MUST identify a closeable session, and it MUST become invalid
+   after a session is closed.
+1. It MUST remain the same as the identified session's state changes.
```
Comment (ritave) [inline CAIPs/caip-171.md:57]: I don't understand this point.

File: CAIPs/caip-171.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+> that it be expressed as a string with few reserve characters, under 256
+> characters in length. This is by design and the value is
+> implementation-dependent.
+
+```typescript
+type SessionToken = string;
+```
+
+Properties of the `sessionIdentifier` are as follows:
+1. It MUST uniquely identify an open and stateful session.
```
Comment (ritave) [inline CAIPs/caip-171.md:54]: Please actually increment the bulled points

File: CAIPs/caip-171.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+```typescript
+type SessionToken = string;
+```
+
+Properties of the `sessionIdentifier` are as follows:
+1. It MUST uniquely identify an open and stateful session. 
+1. It MUST identify a closeable session, and it MUST become invalid
+   after a session is closed.
+1. It MUST remain the same as the identified session's state changes.
+1. It MUST be serializable into JSON. Serialization and later deserialization using
```
Comment (ritave) [inline CAIPs/caip-171.md:58]: `1. It MUST be between inclusive 1 and 256 characters in length.`

File: CAIPs/caip-171.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+JSON MUST result in the same value.
+
+## Copyright
+
+Copyright and related rights waived via
+[CC0](https://creativecommons.org/publicdomain/zero/1.0/).
```
Comment (ritave) [inline CAIPs/caip-171.md:64]: Also I'd suggest having the CC0 in the LICENSE file of this repo as well.

That's what we do in [Snaps Improvement Proposals](https://github.com/MetaMask/SIPs/blob/60c212d1da607005bef989fd364565f21c66f111/sip-template.md?plain=1#L37)

File: CAIPs/caip-25.md [hunk 8]
```diff
@@ -137,9 +137,11 @@ The valid error messages codes are the following:
     * code = 5102
     * message = "Requested events are not supported"
 
-## Links
+## Changelog
 
-n/a
+- 2022-11-26: add mandatory indexing by session identifier (i.e. CAIP-171 requirement)
```
Comment (ritave) [inline CAIPs/caip-25.md:142]: It's a Draft, imho the changelog shouldn't be in it until it's Final, where we could add Errata.

**Review by ritave — APPROVED** [review]
Generally looks good with minor asks

File: CAIPs/caip-171.md (hunk 1 above)
Comment (bumblefudge) [reply CAIPs/caip-171.md:50]: 🤦‍♀️

File: CAIPs/caip-171.md (hunk 4 above)
Comment (bumblefudge) [reply CAIPs/caip-171.md:57]: the identifier remains constant when the state changes, i.e. not content-addressable/hash-based.  better phrasing appreciated!

File: CAIPs/caip-171.md (hunk 3 above)
Comment (bumblefudge) [reply CAIPs/caip-171.md:45]: oops let's leave that in the separate PR so it doesn't block this one

File: CAIPs/caip-171.md (hunk 5 above)
Comment (bumblefudge) [reply CAIPs/caip-171.md:54]: good point, better not to rely on markdown for people that will be reading it in emacs/vim/etc

**Pull request merged**